src/Stack/Setup.hs

```diff
@@ -1085,14 +1085,21 @@ installGHCPosix version downloadInfo _ archiveFile archiveType tempDir destDir =
 
     let runStep step wd env cmd args = do
             menv' <- modifyEnvOverride menv (Map.union env)
-            result <- withWorkingDir wd
-                    $ withEnvOverride menv'
-                    $ withProc cmd args
-                    $ try
-                    . readProcessStdout_
-                    -- Calling the ./configure script requires that stdin is
-                    -- open
-                    . setStdin (useHandleOpen stdin)
+            result <- do
+                let logLines = CL.mapM_ (logDebug . T.decodeUtf8With T.lenientDecode)
```

Comment:
This implementation is different from the one in `logProcessStderrStdout`, which includes a call to `CB.lines`:

```haskell
let logLines = CB.lines .| CL.mapM_ (logInfo . decodeUtf8With lenientDecode)
```

As implemented here, it will not break the output up into individual lines.

Reply:
Another difference is that I have used `concurrently_` since we are ignoring the return values subsequently.